pennylane_lightning/src/algorithms/JacobianTape.hpp

```diff
@@ -274,15 +226,26 @@ template <class T> class JacobianData {
      * @param ps Pointer to the statevector data.
      * @param obs Observables for which to calculate Jacobian.
      * @param ops Operations used to create given state.
-     * @param trainP List of parameters participating in Jacobian
-     * calculation. This must be sorted.
+     * @param trainP Sorted list of parameters participating in Jacobian
+     * computation.
+     *
+     * @rst
+     * Each value :math:`i` in trainable params means that
+     * we want to take a derivative respect to the :math:`i`-th operation.
+     *
+     * Further note that ``ops`` does not contain state preparation operations
+     * (e.g. QubitStateVector) or Hamiltonian coefficients.
+     * @endrst
      */
     JacobianData(size_t num_params, size_t num_elem, std::complex<T> *ps,
-                 std::vector<ObsDatum<T>> obs, OpsData<T> ops,
-                 std::vector<size_t> trainP)
+                 std::vector<std::shared_ptr<Observable<T>>> obs,
```

Comment:
I assume the `shared_ptr` approach lowers the overall transfer overhead here? Or is it due to ownership elsewhere?

Reply:
When we serialize, Python directly creates an `std::shared_ptr<[a subclass of Observable<T>]>` object, e.g. `ob = NamedObsC64("X", [0]` creates `std::shared_ptr<NamedObs<double>>`. Thus we can just use it. It is possible (in principle) to use `unique_ptr` but a Python code should know that the object is consumed by the C++ code (which is rather complicated).